include/boost_plugin_loader/utils.h

```diff
@@ -53,6 +55,13 @@ boost::dll::shared_library loadLibrary(const std::string& library_name, const st
 bool isSymbolAvailable(const std::string& symbol_name, const std::string& library_name,
                        const std::string& library_directory = "");
 
+/**
+ * @brief This will remove libraries with full path in the provided library_names and return them.
+ * @param library_names The set to search and remove libraries with full paths
+ * @return A set of the libraries provided as full path
+ */
+std::set<std::string> extractLibrariesWithFullPath(std::set<std::string>& library_names);
```

Comment:
This function has really has two outputs right?: a set of strings that represents the libraries that are defined by full absolute paths (i.e., the actual return of the current function), and a reduced set of library names that are not fully defined absolute paths (i.e., the modified input passed by reference).

If this is the case, I would recommend changing the signature of the function to be something like this for more clarity:

```suggestion
/**
 * @brief Separates a set of library paths with mixed path specification into one set with libraries that are specified as fully-defined absolute paths and a second set with libraries that are specified by name only.
 * @param library_names The set to search and remove libraries with full paths
 * @return A tuple containing 1) the set of the libraries specified as fully-defined absolute paths, and 2) the set of remaining libraries specified as library names
 */
std::tuple<std::set<std::string>, std::set<std::string>> separateLibraryPathSpecifications(std::set<std::string> library_names);
```

Reply:
The latest reflect these changes.